app/code/core/Mage/Core/Model/Resource/Session.php

```diff
@@ -257,19 +252,18 @@ public function destroy($sessId)
      * Garbage collection
      *
      * @param int $sessMaxLifeTime ignored
-     * @return bool
+     * @return int|false
      */
-    #[\ReturnTypeWillChange]
-    public function gc($sessMaxLifeTime)
+    public function gc($sessMaxLifeTime): int|false
     {
         if ($this->_automaticCleaningFactor > 0) {
             if ($this->_automaticCleaningFactor == 1 ||
                 rand(1, $this->_automaticCleaningFactor) == 1
             ) {
                 $where = ['session_expires < ?' => Varien_Date::toTimestamp(true)];
-                $this->_write->delete($this->_sessionTable, $where);
+                return $this->_write->delete($this->_sessionTable, $where);
             }
         }
-        return true;
+        return false;
```

Comment:
Not sure about this. wouldn't this mean `gc()` has failed? See example in https://www.php.net/manual/en/class.sessionhandlerinterface.php
![image](https://github.com/OpenMage/magento-lts/assets/1106470/9935c21d-5002-456e-ac90-db4a502fb9da)

Reply:
The docs for `gc()` says:
> Returns the number of deleted sessions on success, or false on failure. Note this value is returned internally to PHP for processing.

So it should return 0 instead of `false`.

Reply:
done, thanks!